poly/sharing_new.go

```diff
@@ -0,0 +1,161 @@
+import "github.com/dedis/crypto/abstract"
+
+// XXX: Un/Marshalling functions are still missing.
+
+// PriShare represents an individual private share v = p(i).
+type PriShare struct {
+	I int             // Index of the private share
+	V abstract.Scalar // Value of the private share
+}
+
+// PriPoly represents a secret sharing polynomial.
+type PriPoly struct {
+	g      abstract.Group    // Cryptographic group
+	coeffs []abstract.Scalar // Coefficients of the polynomial
+}
+
+// NewPriPoly creates a new secret sharing polynomial for the cryptographic
+// group g, the secret sharing threshold t, and the secret to be shared s.
+func NewPriPoly(g abstract.Group, t int, s abstract.Scalar) *PriPoly {
+	// XXX: old PriPoly.Pick()
+	return nil
+}
+
+// Threshold returns the secret sharing threshold.
+func (p *PriPoly) Threshold() int {
+	return len(p.coeffs)
+}
+
+// SharedSecret returns the shared secret p(0), i.e., the constant term of the polynomial.
+func (p *PriPoly) SharedSecret() abstract.Scalar {
```

Comment:
The shared secret is already contained in the `coeffs` slice at index 0, so this function just returns that value. And the current constructor for a `PriPoly` allows you to also give no specific secret at which point it is just generated randomly, see https://github.com/dedis/crypto/blob/master/poly/sharing.go#L40-L41. I think we want to keep that functionality.